File: ConsoleApp1/ConsoleApp1/DauSachGiaoTrinh.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QLTV
{
    class DauSachGiaoTrinh:DauSach
    {
        protected string sBoMon;
		protected List<QuyenSachGiaoTrinh> lDSQuyenSach;
		public string BoMon
		{
			get { return this.sBoMon; }
		}
		public List<QuyenSachGiaoTrinh> DSQuyenSach
		{
			get { return this.lDSQuyenSach; }
			set { this.lDSQuyenSach = value; }
		}
		public DauSachGiaoTrinh()
		{
			this.XacDinhThue();
			this.lDSQuyenSach = new List<QuyenSachGiaoTrinh>();
		}
		
        public DauSachGiaoTrinh(string MaDauSach, string Tua, string NXB, string TacGia, double Gia, string BoMon) : base(MaDauSach, Tua, NXB, TacGia, Gia)
        {
            this.sBoMon = BoMon;
			this.lDSQuyenSach = new List<QuyenSachGiaoTrinh>();
		}
      
		public override void Nhap()
		{
			base.Nhap();
			Console.Write(" Bo mon: ");
			this.sBoMon = Console.ReadLine();
		}
		public override void Xuat()
		{
			base.Xuat();
			Console.WriteLine(" Bo mon: " + this.sBoMon);
		}
		public override void TinhThanhTien()
		{
		}
		public override void XacDinhThue()
		{
			this.dThue = 10;
		}
	}
}

File: Quan_Ly_Thu_Vien/src/DauSachGiaoTrinh.java
import java.util.ArrayList;
import java.util.Scanner;

public class DauSachGiaoTrinh extends DauSach
{
	protected String sBoMon;
	protected ArrayList <QuyenSachGiaoTrinh> lDSQuyenSach;
	
	public String getsBoMon() 
	{
		return sBoMon;
	}

	public void setsBoMon(String sChuyenNganh)
	{
		this.sBoMon = sChuyenNganh;
	}

	public ArrayList<QuyenSachGiaoTrinh> getlDSQuyenSach() 
	{
		return lDSQuyenSach;
	}

	public void setlDSQuyenSach(ArrayList<QuyenSachGiaoTrinh> lDSQuyenSach)
	{
		this.lDSQuyenSach = lDSQuyenSach;
	}
	public DauSachGiaoTrinh()
	{
		XacDinhThue();
		this.lDSQuyenSach = new ArrayList<QuyenSachGiaoTrinh>();
	}
	public DauSachGiaoTrinh(String MaSach,String Tua,String NhaXuatBan, String TacGia, double DonGia,String BoMon)
	{
		super(MaSach,Tua, NhaXuatBan,TacGia, DonGia);
		this.sBoMon = BoMon;
		XacDinhThue();
		this.lDSQuyenSach = new ArrayList<QuyenSachGiaoTrinh>();
	}
	
	@SuppressWarnings("resource")
	@Override
	public void Nhap()
	{
		super.Nhap();
		Scanner sc = new Scanner(System.in);		
		System.out.print("Nhap bo mon: ");
		this.sBoMon = sc.nextLine();
	}	
	@Override
	public void Xuat()
	{
		super.Xuat();
		System.out.println(" Bo mon: " + this.sBoMon);
	}
	@Override 
	public void TinhThanhTien()
	{
	}
	@Override
	public void XacDinhThue()
	{
		this.dThue= 10;
	}
}
